sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs

```diff
@@ -89,11 +53,10 @@ protected override void Initialize(SonarAnalysisContext context)
                     }
 
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
-                    var attributes = methodSymbol?.GetAttributes();
                     if (methodSymbol == null ||
-                        !IsTestMethod(attributes) ||
-                        HasExpectedExceptionAttribute(attributes) ||
-                        IsTestIgnored(attributes))
+                        !methodSymbol.IsTestMethod() ||
+                        methodSymbol.HasExpectedExceptionAttribute() ||
+                        IsTestIgnored(methodSymbol))
```

Comment:
The new code will be marginally less performant since each extension method calls _methodSymbol.GetAttributes_. I think this is an acceptable price to pay for having common behaviour across the rules.